routing/payment_session_source.go

```diff
@@ -42,6 +43,8 @@ type SessionSource struct {
 	// PathFindingConfig defines global parameters that control the
 	// trade-off in path finding between fees and probability.
 	PathFindingConfig PathFindingConfig
+
+	TrafficShaper fn.Option[TlvTrafficShaper]
```

Comment:
nit: needs Godoc comment.
Can copy the one from the interface definition itself:
```
// TlvTrafficShaper is an interface that allows the sender to determine if a
// payment should be carried by a channel based on the TLV records that may be
// present in the `update_add_htlc` message or the channel commitment itself.
```